src/java/com/vaklinov/zcashui/AddressesPanel.java

```diff
@@ -376,6 +383,35 @@ private JTable createAddressBalanceTable(String rowData[][])
 
 		for (String address : tAddressesCombined)
 		{
+			String addressToDisplay = address;
+			// Make sure the current address is not watch-only or invalid
+			if (!this.validationMap.containsKey(address))
+			{
+				boolean validationResult = this.clientCaller.isWatchOnlyOrInvalidAddress(address);
+				this.validationMap.put(address, new Boolean(validationResult));	
+				
+				if (validationResult)
+				{
+		            JOptionPane.showMessageDialog(
```

Comment:
Just a note that in this `if` block, you're mixing tabs and spaces

I might go through and normalize this throughout the project so I don't think it _needs_ to be changed, but an FYI

Reply:
It's mixed in every file. :)